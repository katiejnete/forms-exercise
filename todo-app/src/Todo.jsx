import React from "react";

const Todo = ({ task, removeTodo, id }) => {
  return (
    <div data-testid="task">
      {task}
      <button onClick={() => removeTodo(id)}>X</button>
    </div>
  );
};

export default Todo;
