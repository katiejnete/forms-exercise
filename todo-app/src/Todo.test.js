import React from "react";
import {render} from "@testing-library/react";
import Todo from "./Todo.jsx";
import removeTodo from "./TodoList.jsx"

// smoke test
it("renders without crashing", () => {
    render(<Todo key="ranUuid" id="ranUuid" removeTodo={removeTodo} task="task"/>);
})

// snapshot test
it("matches snapshot", () => {
    const {asFragment} = render(<Todo key="ranUuid" id="ranUuid" removeTodo={removeTodo} task="task"/>);
    expect(asFragment()).toMatchSnapshot();
})