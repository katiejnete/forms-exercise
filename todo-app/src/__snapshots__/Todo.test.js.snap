// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="task"
  >
    task
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
